src/grt/src/fastroute/src/maze.cpp

```diff
@@ -1174,6 +1167,91 @@ void FastRouteCore::mazeRouteMSMD(const int iter,
 
   std::vector<bool> pop_heap2(y_grid_ * x_range_, false);
 
+  auto updateAdjacent = [&](const int cur_x,
```

Comment:
Add a small comment with the summary of what this lambda function is doing. Here and below.